cloudferrylib/os/discovery/model.py

```diff
@@ -419,29 +518,45 @@ class Reference(fields.Field):
     def __init__(self, model_class, many=False, ensure_existence=False,
                  **kwargs):
         super(Reference, self).__init__(**kwargs)
-        self.model_class = model_class
+        if isinstance(model_class, basestring):
+            self._model_class_name = model_class
+            self._model_class = None
+        else:
+            self._model_class = model_class
         self.many = many
         self.ensure_existence = ensure_existence
 
+    @property
+    def model_class(self):
+        if self._model_class is None:
+            self._model_class = importutils.import_class(
+                self._model_class_name)
+        return self._model_class
+
     def _serialize(self, value, attr, obj):
         if value is None:
             return None
         model = self.model_class
         pk_field = model.pk_field
         if self.many:
-            return [_object_id_to_dict(model, getattr(x, pk_field))
+            return [getattr(x, pk_field).to_dict(model)
                     for x in value]
         else:
-            return _object_id_to_dict(model, getattr(value, pk_field))
+            return getattr(value, pk_field).to_dict(model)
```

Comment:
It is custom Marshmallow field, `_serialize` and `_deserialize` are designed to work with JSON-like objects (lists and dictionaries fall into this category). More info: http://marshmallow.readthedocs.org/en/latest/custom_fields.html